skorch/callbacks/logging.py

```diff
@@ -116,97 +117,171 @@ class NeptuneLogger(Callback):
-    ... neptune_logger.experiment.log_artifact('basic_model.pkl')
+    ... neptune_logger.run["basic_model"].upload(File('basic_model.pkl'))
 
-    >>> # close experiment
-    ... neptune_logger.experiment.stop()
+    >>> # Close the run
+    ... neptune_logger.run.stop()
 
     Parameters
     ----------
-    experiment : neptune.experiments.Experiment
-      Instantiated ``Experiment`` class.
+    run : neptune.new.Run
+      Instantiated ``Run`` class.
 
     log_on_batch_end : bool (default=False)
       Whether to log loss and other metrics on batch level.
 
     close_after_train : bool (default=True)
-      Whether to close the ``Experiment`` object once training
+      Whether to close the ``Run`` object once training
       finishes. Set this parameter to False if you want to continue
-      logging to the same Experiment or if you use it as a context
+      logging to the same run or if you use it as a context
       manager.
 
     keys_ignored : str or list of str (default=None)
       Key or list of keys that should not be logged to Neptune. Note that in
       addition to the keys provided by the user, keys such as those starting
       with ``'event_'`` or ending on ``'_best'`` are ignored by default.
 
+    base_namespace: str
+      Namespace (folder) under which all metadata logged by the ``NeptuneLogger``
+      will be stored. Defaults to "training".
+
     Attributes
     ----------
-    first_batch_ : bool
-        Helper attribute that is set to True at initialization and changes
-        to False on first batch end. Can be used when we want to log things
-        exactly once.
-
     .. _Neptune: https://www.neptune.ai
 
     """
 
     def __init__(
             self,
-            experiment,
+            run,
             log_on_batch_end=False,
             close_after_train=True,
             keys_ignored=None,
+            base_namespace='training',
     ):
-        self.experiment = experiment
+        self.run = run
         self.log_on_batch_end = log_on_batch_end
         self.close_after_train = close_after_train
         self.keys_ignored = keys_ignored
 
-    def initialize(self):
-        self.first_batch_ = True
+        if base_namespace.endswith("/"):
+            self._base_namespace = base_namespace[:-1]
+        else:
+            self._base_namespace = base_namespace
 
+    @property
+    def _metric_logger(self):
+        return self.run[self._base_namespace]
+
+    @staticmethod
+    def _get_obj_name(obj):
+        return type(obj).__name__
+
+    def initialize(self):
         keys_ignored = self.keys_ignored
         if isinstance(keys_ignored, str):
             keys_ignored = [keys_ignored]
         self.keys_ignored_ = set(keys_ignored or [])
         self.keys_ignored_.add('batches')
         return self
 
+    def on_train_begin(self, net, X, y, **kwargs):
+
+        self._metric_logger['model/model_type'] = NeptuneLogger._get_obj_name(net.module_)
```

Comment:
Can we change all uses of `NeptuneLogger._get_obj_name` to `self._get_obj_name`? Same for `_model_summary_file`. For static methods, it doesn't really matter, but I think that `self` is more intuitive here and if it ever needs to be changed to a bound method, these calls don't need to be edited.

Reply:
Sure, will do.